lib/field_names.cpp

```diff
@@ -48,15 +48,22 @@ FieldNames::init(const ResultBase* res)
 }
 
 
+#if __cplusplus >= 201103L
+unsigned int
+FieldNames::operator [](std::string s) const
+{
+    std::string temp1(std::move(s));
+#else
 unsigned int
 FieldNames::operator [](const std::string& s) const
 {
 	std::string temp1(s);
+#endif
 	internal::str_to_lwr(temp1);
 	for (const_iterator it = begin(); it != end(); ++it) {
 	std::string temp2(*it);
 		internal::str_to_lwr(temp2);
-		if (temp2.compare(temp1) == 0) {
+		if (temp2 == temp1) {
```

Comment:
[That's C++17+ only](https://en.cppreference.com/w/cpp/string/basic_string_view/operator_cmp). Besides which, I doubt it makes a difference in the resulting assembly.

Reply:
Wait, but `compare(const std::string&)` and  `operator==(const std::string&, const std::string&)` have no difference in result code?
P.S. I made this change purely for readability and do not insist on it.